source/AI.cpp

```diff
@@ -405,15 +419,22 @@ void AI::UpdateKeys(PlayerInfo &player, Command &activeCommands)
 	// Get rid of any invalid orders. Carried ships will retain orders in case they are deployed.
 	for(auto it = orders.begin(); it != orders.end(); )
 	{
-		if(it->second.type & Orders::REQUIRES_TARGET)
+		if(it->second.type == Orders::MINE && it->first->Cargo().Free() && it->second.targetAsteroid.expired())
+			it->second.type = Orders::HARVEST;
```

Comment:
It doesn't make sense to not have it harvest after an asteroid breaks up.  Imagine you direct your fleet around the system to mine asteroids but they don't actually pick them up.  Now you, as a player, need to run around and chase the flotsam before it expires.  This is a very strange behavior to want out of the box.

I don't want it to either be disabled or a preference.  It's a core mechanic to the fleet mining.  Without an asteroid blowing it it made sense to have a separate key for toggling harvest but for mining I see it as a requirement for good gameplay.

Reply:
Can you make it send the message ("Your fleet is preparing to harvest...") when they automatically switch to that mode (if they aren't already) to avoid confusion as to why pressing the harvest command actually turns it off now even though hte player didn't turn it on themselves?

Reply:
Going through the code I do not think this is feasible without spam.

This transition is not your whole fleet.  It is an individual ship.  So if you order a single ship or three ships each will transition from mining to harvest mode individually (not as a group).  There's no easy way to make a unified message without inserting some kind of counter and doing complex message building similar to IssueOrders.

Reply:
We could just have a "tutorial" that fires up once saying the ships will collect automatically so nobody is surprised?

Reply:
Another tutorial could be easy to add.

Reply:
@warp-core @Amazinite If that's good enough I can add it.  i.e. the first time a player breaks up an asteroid with an escort;
```
help "escorts mining asteroids"
	`Your escort will transition to harvesting flotsam after the targeted asteroid is destroyed.`
	``
	`You can get more information about an asteroid by equipping tactical scanning with asteroid scanning.`
```
This would drop a bread crumb for the player that a tactical scanner works and alerts the player that their escorts transition to harvest mode automatically.

Reply:
To be clear I do not plan to do anything for this PR.  This PR is old and large.  It might be better to open a followup PR to address this. After this PR is merged.